tests/certs_test.go

```diff
@@ -1,28 +1,476 @@
+					// is a wildcard domain ('*.foo.com') and we don't want Gomega
+					// interpreting this string as a regexp
+					var substring string
+
+					sess, err := start("deis domains:add %s --app=%s", domain, testApp.Name)
+					Expect(err).NotTo(HaveOccurred())
+					substring = fmt.Sprintf("Adding %s to %s...", domain, testApp.Name)
+					Eventually(sess.Wait().Out.Contents()).Should(ContainSubstring(substring))
+					Eventually(sess).Should(Say("done"))
+					Eventually(sess).Should(Exit(0))
+
+					sess, err = start("deis certs:add %s %s %s", certName, cert, key)
+					Expect(err).NotTo(HaveOccurred())
+					Eventually(sess).Should(Say("Adding SSL endpoint..."))
+					Eventually(sess).Should(Say("done"))
+					Eventually(sess).Should(Exit(0))
+
+					sess, err = start("deis certs:attach %s %s", certName, domain)
+					Expect(err).NotTo(HaveOccurred())
+					substring = fmt.Sprintf("Attaching certificate %s to domain %s...", certName, domain)
+					Eventually(sess.Wait().Out.Contents()).Should(ContainSubstring(substring))
+					Eventually(sess).Should(Say("done"))
+					Eventually(sess).Should(Exit(0))
+
+					sess, err = start("deis certs:info %s", certName)
+					Expect(err).NotTo(HaveOccurred())
+					Eventually(sess).Should(Say("=== %s Certificate", certName))
+					Eventually(sess.Wait().Out.Contents()).Should(ContainSubstring(domain))
+					Eventually(sess).Should(Exit(0))
+
+					sess, err = start("deis certs:detach %s %s", certName, domain)
+					Expect(err).NotTo(HaveOccurred())
+					substring = fmt.Sprintf("Detaching certificate %s from domain %s...", certName, domain)
+					Eventually(sess.Wait().Out.Contents()).Should(ContainSubstring(substring))
+					Eventually(sess).Should(Say("done"))
+					Eventually(sess).Should(Exit(0))
+
+					sess, err = start("deis certs:info %s", certName)
+					Expect(err).NotTo(HaveOccurred())
+					Eventually(sess).Should(Say("=== %s Certificate", certName))
+					Eventually(sess).Should(Say("No connected domains"))
+					Eventually(sess).Should(Exit(0))
+
+					sess, err = start("deis domains:remove %s --app=%s", domain, testApp.Name)
+					Expect(err).NotTo(HaveOccurred())
+					substring = fmt.Sprintf("Removing %s from %s...", domain, testApp.Name)
+					Eventually(sess.Wait().Out.Contents()).Should(ContainSubstring(substring))
+					Eventually(sess).Should(Say("done"))
+					Eventually(sess).Should(Exit(0))
+
+					removeCerts([]string{certName})
+				},
+
+				Entry("a non-wildcard cert to a wildcard domain",
+					domains["wildcard"], certs["www"].Name, certs["www"].CertPath, certs["www"].KeyPath),
+				Entry("a non-wildcard cert to a non-wildcard domain",
+					domains["foo"], certs["foo"].Name, certs["foo"].CertPath, certs["foo"].KeyPath),
+				Entry("a non-wildcard cert to a non-wildcard domain",
+					domains["bar"], certs["bar"].Name, certs["bar"].CertPath, certs["bar"].KeyPath),
+			)
+
+			DescribeTable("2 certs (1 wildcard), 3 domains (1 wildcard)",
```

Comment:
I think it's fine as-is if it's testing the use cases we want. If you feel like the other approach is better then by all means go for it, but I live by the mantra of "Ship first, refactor later". :)